src/Kubernetes.WebSocket.cs

```diff
@@ -0,0 +1,144 @@
+
+        /// <inheritdoc/>
+        public Task<WebSocket> WebSocketNamespacedPodExecWithHttpMessagesAsync(string name, string @namespace = "default", string command = "/bin/sh", string container = null, bool stderr = true, bool stdin = true, bool stdout = true, bool tty = true, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (@namespace == null)
+            {
+                throw new ArgumentNullException(nameof(@namespace));
+            }
+
+            if (command == null)
+            {
+                throw new ArgumentNullException(nameof(command));
+            }
+
+            // Tracing
+            bool _shouldTrace = ServiceClientTracing.IsEnabled;
+            string _invocationId = null;
+            if (_shouldTrace)
+            {
+                _invocationId = ServiceClientTracing.NextInvocationId.ToString();
+                Dictionary<string, object> tracingParameters = new Dictionary<string, object>();
+                tracingParameters.Add("command", command);
+                tracingParameters.Add("container", container);
+                tracingParameters.Add("name", name);
+                tracingParameters.Add("namespace", @namespace);
+                tracingParameters.Add("stderr", stderr);
+                tracingParameters.Add("stdin", stdin);
+                tracingParameters.Add("stdout", stdout);
+                tracingParameters.Add("tty", tty);
+                tracingParameters.Add("cancellationToken", cancellationToken);
+                ServiceClientTracing.Enter(_invocationId, this, nameof(WebSocketNamespacedPodExecWithHttpMessagesAsync), tracingParameters);
+            }
+
+            // Construct URL
+            var uriBuilder = new UriBuilder(BaseUri);
+            uriBuilder.Scheme = BaseUri.Scheme == "https" ? "wss" : "ws";
+
+            if (!uriBuilder.Path.EndsWith("/"))
+            {
+                uriBuilder.Path += "/";
+            }
+
+            uriBuilder.Path += $"api/v1/namespaces/{@namespace}/pods/{name}/exec";
+
+            List<string> _queryParameters = new List<string>();
+            _queryParameters.Add(string.Format("command={0}", Uri.EscapeDataString(command)));
+
+            if (container != null)
+            {
+                _queryParameters.Add(string.Format("container={0}", Uri.EscapeDataString(container)));
+            }
+
+            _queryParameters.Add(string.Format("stderr={0}", stderr ? 1 : 0));
+            _queryParameters.Add(string.Format("stdin={0}", stdin ? 1 : 0));
+            _queryParameters.Add(string.Format("stdout={0}", stdout ? 1 : 0));
+            _queryParameters.Add(string.Format("tty={0}", tty ? 1 : 0));
+
+            uriBuilder.Query = string.Join("&", _queryParameters);
```

Comment:
take look at `QueryHelpers.AddQueryString` I already imported this.
it might help work encode strings

Reply:
Like this:

```csharp
string queryString = string.Empty;
queryString = QueryHelpers.AddQueryString(queryString, "command", command);

if (container != null)
{
    queryString = QueryHelpers.AddQueryString(queryString, "container", container);
}

queryString = QueryHelpers.AddQueryString(queryString, "stderr", stderr ? "1" : "0");
queryString = QueryHelpers.AddQueryString(queryString, "stdin", stdin ? "1" : "0");
queryString = QueryHelpers.AddQueryString(queryString, "stdout", stdout ? "1" : "0");
queryString = QueryHelpers.AddQueryString(queryString, "tty", tty ? "1" : "0");

uriBuilder.Query = queryString;
```
?

Reply:
```
            var queryString = QueryHelpers.AddQueryString(string.Empty, new Dictionary<string, string>
            {
                { "command", "aaaa"},
                { "tty", "1"}
            });
```

Reply:
Done, thanks!